cylc/uiserver/resolvers.py

```diff
@@ -19,48 +19,42 @@
 from getpass import getuser
 import os
 from copy import deepcopy
-from functools import partial
 from subprocess import Popen, PIPE, DEVNULL
-from types import SimpleNamespace
 from typing import (
-    TYPE_CHECKING, Any, Callable, Dict, Iterable, List, Union
+    TYPE_CHECKING, Any, Callable, Dict, Iterable, List, Tuple, Union
 )
 
 from graphql.language.base import print_ast
 
 from cylc.flow.data_store_mgr import WORKFLOW
-from cylc.flow.exceptions import CylcError, ServiceFileError
+from cylc.flow.exceptions import (
+    ServiceFileError,
+    WorkflowFilesError,
+)
 from cylc.flow.network.resolvers import BaseResolvers
+from cylc.flow.scripts.clean import CleanOptions
 from cylc.flow.workflow_files import init_clean
 
-
-class InvalidSchemaOptionError(CylcError):
-    ...
-
-
 if TYPE_CHECKING:
+    from concurrent.futures import Executor
     from logging import Logger
+    from optparse import Values
     from graphql import ResolveInfo
     from cylc.flow.data_store_mgr import DataStoreMgr
     from cylc.flow.id import Tokens
+    from cylc.flow.option_parsers import Options
     from cylc.uiserver.workflows_mgr import WorkflowsManager
 
 
 # show traceback from cylc commands
 DEBUG = True
-CLEAN = 'clean'
-OPT_CONVERTERS: Dict[str, Dict[str, Union[Callable, None]]] = {
-    CLEAN: {
-        'rm': lambda opt, value: ('rm_dirs', [value]),
```

Comment:
As Oliver commented in #368 
> This resulted in `['']` which is "Truthy" effectively disabling the clean operation.

Reply:
That's vaguely amusing. A bag of nothing is falsey, but a bag containing a single nothing is truthy.

Reply:
At least it's not JS where a bag of nothing is truthy!

Reply:
Because:

```python
>>> len('')
0
>>> len([])
0
>>> len([''])
1
```